### PR TITLE
kvserver/idalloc: rename confusing var

### PR DESCRIPTION
**Pull request by andreimatei**
An error with a very specific meaning was shadowing a broader err,
making the code very hard to read. It looked like any failure is causing
the idallocator worker routine to exit, whereas only an error caused by
the server quiescing had that effect.

Release note: None

### CODE REVIEW

**Comment by andreimatei** [comment]
cc @jasobrown

**Review by nvb — APPROVED** [review]

File: pkg/kv/kvserver/idalloc/id_alloc.go [hunk 1]
```diff
@@ -91,10 +92,9 @@ func (ia *Allocator) start() {
 			var res kv.KeyValue
 			for r := retry.Start(base.DefaultRetryOptions()); r.Next(); {
 				idKey := ia.idKey.Load().(roachpb.Key)
-				if err := ia.stopper.RunTask(ctx, "storage.Allocator: allocating block", func(ctx context.Context) {
+				if stopperErr := ia.stopper.RunTask(ctx, "idallocator: allocating block", func(ctx context.Context) {
```
Comment (nvanbenschoten) [inline pkg/kv/kvserver/idalloc/id_alloc.go]: `s/idallocator/idalloc/`? to remain consistent with the package name?

Comment (andreimatei) [reply pkg/kv/kvserver/idalloc/id_alloc.go]: done

**Review by knz — APPROVED** [review]

**Pull request merged**